tests/plugins/selection.test.ts

```diff
@@ -1010,57 +1012,97 @@ describe("move elements(s)", () => {
+              A6: { content : "a" }
+            },
+            merges: ["C3:D3"],
+            styles: { 1: { textColor: "#fe0000" } },
           },
-          merges: ["C3:D3"],
-          styles: { 1: { textColor: "#fe0000" } },
-        },
-      ],
+        ],
+      });
     });
-  });
 
-  test.each([
-    ["B2", ["B2:E4", "A1:J10", "B2"]],
-    ["A2", ["A2:E4", "A1:J10", "A2"]],
-    ["B1", ["B1:E4", "A1:J10", "B1"]],
-    ["E3", ["B2:E4", "A1:J10", "E3"]],
-    ["A1", ["A1:J10", "A1"]],
-    ["A6", ["A1:J10", "A6"]],
-  ])("Selection loop with anchor %s", (anchor: string, expectedZones: string[]) => {
-    selectCell(model, anchor);
-    for (const zone of expectedZones) {
-      model.selection.loopSelection();
+    test.each([
+      ["B2", ["B2:E4", "A1:J10", "B2"]],
+      ["A2", ["A2:E4", "A1:J10", "A2"]],
+      ["B1", ["B1:E4", "A1:J10", "B1"]],
+      ["E3", ["B2:E4", "A1:J10", "E3"]],
+      ["A1", ["A1:J10", "A1"]],
+      ["A6", ["A1:J10", "A6"]],
+    ])("Selection loop with anchor %s", (anchor: string, expectedZones: string[]) => {
+      selectCell(model, anchor);
+      for (const zone of expectedZones) {
+        model.selection.loopSelection();
+        const selection = model.getters.getSelectedZone();
+        expect(zoneToXc(selection)).toEqual(zone);
+        expect(zoneToXc(toZone(getActivePosition(model)))).toEqual(anchor);
+      }
+    });
+
+    test.each([
+      ["B2", "B2:E4"],
+      ["A2", "A2:E4"],
+      ["B1", "B1:E4"],
+      ["E3", "B2:E4"],
+      ["A1", "A1"],
+      ["A6", "A6"],
+    ])("Select table around the anchor %s", (anchor: string, expectedZone: string) => {
+      selectCell(model, anchor);
+      model.selection.selectTableAroundSelection();
       const selection = model.getters.getSelectedZone();
-      expect(zoneToXc(selection)).toEqual(zone);
-      expect(zoneToXc(toZone(getActivePosition(model)))).toEqual(anchor);
-    }
+      expect(zoneToXc(selection)).toEqual(expectedZone);
+      expect(zoneToXc(positionToZone(model.getters.getActivePosition()))).toEqual(anchor);
+    });
   });
 
-  test.each([
-    ["B2", "B2:E4"],
-    ["A2", "A2:E4"],
-    ["B1", "B1:E4"],
-    ["E3", "B2:E4"],
-    ["A1", "A1"],
-    ["A6", "A6"],
-  ])("Select table around the anchor %s", (anchor: string, expectedZone: string) => {
-    selectCell(model, anchor);
-    model.selection.selectTableAroundSelection();
-    const selection = model.getters.getSelectedZone();
-    expect(zoneToXc(selection)).toEqual(expectedZone);
-    expect(zoneToXc(positionToZone(model.getters.getActivePosition()))).toEqual(anchor);
+  describe("Viewport doesn't move", () => {
+    let model: Model;
+    beforeEach(() => {
+      model = new Model();
+      setCellContent(model, "A1", "a");
+      setCellContent(model, "A2", "a");
+      setCellContent(model, "A3", "a");
+      setCellContent(model, "A4", "a");
+
+      selectCell(model, "A4");
+      setViewportOffset(model, 0, DEFAULT_CELL_HEIGHT);
```

Comment:
please don't. create only empty things in before each.

It makes tests unreadable because of the noise and un-debuggable again because of the noise. Tests are strongly coupled to `beforeEach` (and are no longer that independent)
It might be ok-ish right now because there's only two tests. But people will add more and more tests, people will copy this (bad) practice, etc.

Reply:
I mean I don't want to copy 8 lines for each tests, `beforeEach` exist for that. I can create an helper function taht does the `setCellContent` and call it at the start of each test, but that's sensibly the same as `beforeEach`

Reply:
> I mean I don't want to copy 8 lines for each tests, beforeEach exist for that. I can create an helper function taht does the setCellContent and call it at the start of each test, but that's sensibly the same as beforeEach

I agree with you it's sensibly the same if you add a getter and use it everywhere. It's still slightly better since it won't "force" the `beforeEach` setup for all future tests if some are to be added here.)

However, what I may have missed to explain properly in my original comment is that test setup should be minimal for the feature under test. No data that doesn't have a real purpose. Do you need *4 cells* to show the feature is working here ? It feels like 2 cells are enough.

The reasons are still the same: no useless noise to read, to debug, to understand what the test is testing